### DISCUSSION TITLE
Cypress unable to manipulate/use array once captured

### DISCUSSION THREAD

**Discussion by LouisGinger** [cypress-io/cypress#18157] [Questions and Help] [answered]
I have a function that captures and array of text from onscreen elements. Cy.log() will show the array contents in the console but after that seems that the array can't be used someting like cy.log(myArray[0]) returns no value and operations like array.join() seems to do nothing too.
Here is my code:
`        
cy.get('parentClass')
        .each(($asset, index, $assets) => {
            const myArray = new Array()
            cy.get($asset).find(childClass1).then(($a) => {AssetCardText.push($a.text())})
            cy.get($asset).find(childClass2).then(($b) => {AssetCardText.push($b.text())})
            cy.get($asset).find(childClass3).then(($c) => {AssetCardText.push($c.text())})
            cy.get($asset).find(childClass4).then(($d) => {AssetCardText.push($d.text())})
            cy.log(myArray) //shows 4 strings in array
            let first = myArray[0]
            cy.log(first)  //shows nothing
        })`

**Comment by bahmutov** [comment, marked as answer]
Watch the explanation in these videos:
- "Why cy.log prints null or undefined" https://www.youtube.com/watch?v=uGWWoq0ECY8
- "When To Use Cypress .Then Callback To Use The Value" https://youtu.be/-aptS3yvqcc

**Reply by LouisGinger** [reply]
wrapping it in a ,then() did the trick. Many thanks. Previously c# selenium so still getting to grips with this!
